src/backend/parser/parse_expr.c

```diff
@@ -1539,6 +1539,15 @@ transformFuncCall(ParseState *pstate, FuncCall *fn)
             targs = ExpandChecksumStar(pstate, fn, fn->location);
 	}
 
+	/*
+	 * Now we give the Post Func ref hook to check whether the function matches any of the Geospatial functions and handle them separately
+	 * STAsText(), STAsBinary(), STDistance()
+	 */
+	if (pstate->p_post_funcref_hook != NULL)
+	{
+		targs = pstate->p_post_funcref_hook(pstate, fn, targs);
+	}
```

Comment:
I think this hook will be useful. Please create a JIRA for moving the checksum code under this hook and assign it to the owner.

Reply:
I will do that once this is merged

Reply:
If we do the change in ANTLR then we wont require a hook

Reply:
Actually we require catalogue lookup for these queries so we can't do these changes in parser side that's why we chose the analyser phase for these handlings.